traits/trait_types.py

```diff
@@ -409,7 +407,7 @@ class Complex(BaseComplex):
     """
 
     #: The C-level fast validator to use:
-    fast_validate = complex_fast_validate
```

Comment:
> are we keeping it around mainly because it is part of the public API

No, I'm just keeping them around to keep this PR simple and focused. I do indeed plan a follow-up PR that removes all three of `int_fast_validate`, `float_fast_validate` and `complex_fast_validate`. They're _not_ part of the public API - they're not exposed in any `api` module. That alone should mean that we don't need to justify removing them, but I did also check all projects available to me to confirm that nothing I can see is using any of these.

I confess I was hoping to go further and remove the use of `numpy` in `trait_types`, but the `Bool` trait messes that up.